project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java

```diff
@@ -37,6 +45,74 @@
 
     private static final String OS = System.getProperty("os.name").toLowerCase(Locale.getDefault());
 
+    public static BuildProject loadBuildProject(Path projectPath) {
+        BuildOptions buildOptions = new BuildOptionsBuilder().offline(true).build();
+        return BuildProject.load(projectPath, buildOptions);
+    }
+
+    static BuildProject loadBuildProject(Path projectPath, BuildOptions options) {
+        BuildOptions buildOptions = new BuildOptionsBuilder().offline(true).build();
+        options.acceptTheirs(buildOptions);
```

Comment:
Here we are mutating the buildOptions. Can we return a new instance with the method `options.acceptTheirs(buildOptions);`?

Reply:
I think we need to fix this issue 
https://github.com/ballerina-platform/ballerina-lang/issues/32499

This is a fundamental issue/.

Reply:
Resolved with 5001c23